code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -347,15 +347,15 @@ var/global/list/damage_icon_parts = list()
 	var/obj/item/organ/internal/eyes/eyes = get_organ(species.vision_organ || BP_EYES)
 	icon_key += istype(eyes) ? eyes.eye_colour : COLOR_BLACK
 
-	for(var/organ_tag in species.has_limbs)
+	for(var/organ_tag in species.has_limbs) // This means non-species limb slots won't draw, todo
 		var/obj/item/organ/external/part = get_organ(organ_tag)
-		if(isnull(part) || part.is_stump() || part.organ_tag == BP_TAIL)
+		if(isnull(part) || part.organ_tag == BP_TAIL)
 			icon_key += "0"
 			continue
 		for(var/M in part.markings)
 			icon_key += "[M][part.markings[M]]"
 		if(part)
-			icon_key += "[part.bodytype.get_icon_cache_uid(part.owner)][part.render_alpha]"
+			icon_key += "[part.bodytype?.get_icon_cache_uid(part.owner)][part.render_alpha]"
```

Comment:
I'm a bit against ignoring the bodytype being null there since its a major fault and probably should crash. 

Maybe we should define some expectations for when update_body should run? Because In my case my expectation were that all limbs must be initialized when its called. That's why I haven't been just blanket calling it each times something changes in the limb hierarchy. Its because otherwise you end up with limbs silently not updating along with the rest because of just timing, and the state after update_body runs now becomes uncertain and unreliable.

Reply:
I can probably remove this change, it was before I worked out why things were double-updating during removal.

Reply:
Yep, fixed.

Reply:
Great, thanks!